src/protocol_splitter.cpp

```diff
@@ -441,13 +450,10 @@ ssize_t DevSocket::write()
 		return payload_len;
 	}
 
-
-	memcpy(_buffer, Sp2HeaderMagic, 3);
-	_buffer[3] = (uint8_t) _type;
-	_buffer[4] = (uint8_t) ((payload_len >> 8) & 0xff);
-	_buffer[5] = (uint8_t) (payload_len & 0xff);
-	_buffer[6] = 0; // reserved
-	_buffer[7] = 0; // reserved
+	_buffer[0] = (uint8_t) Sp2HeaderMagic;
+	_buffer[1] = (uint8_t) (_type & 0x80) | ((payload_len >> 8) & 0x7f);
+	_buffer[2] = (uint8_t) (payload_len & 0xff);
+	_buffer[3] = (_buffer[1] + _buffer[2]) & 0xff; // Checksum
```

Comment:
A bit better would be to just use xor as simple checksum:
```suggestion
	_buffer[3] = _buffer[1] ^ _buffer[2]; // Checksum
```

Reply:
Right, XOR is even simpler implementation. I'll change.

Reply:
Checksum change in both agent_protocol_splitter and PX4-Autopilot PR